packages/backend-core/src/api/resources/OauthAccessToken.ts

```diff
@@ -0,0 +1,23 @@
+import type { OauthAccessTokenJSON } from './JSON';
+
+export class OauthAccessToken {
+  constructor(
+    readonly provider: string,
+    readonly token: string,
+    readonly publicMetadata: Record<string, unknown> = {},
+    readonly label: string,
+    readonly scopes?: string[],
+    readonly tokenSecret?: string,
+  ) {}
+
+  static fromJSON(data: OauthAccessTokenJSON) {
+    return new OauthAccessToken(
+      data.provider,
+      data.token,
+      data.public_metadata,
+      data.label,
+      data.scopes,
+      data.tokenSecret,
```

Comment:
:upside_down_face: Just a note in case the `tokenSecret` camel casing should be snake cased `token_secret` instead.

Reply:
Good catch! Fixed.